### PR TITLE
Add Conversion.into method

### PR DESCRIPTION
**Pull request by odersky**
Small upgrade to make conversions easier to use explicitly.

### CODE REVIEW

**Comment by SethTisue** [comment]
was `as` considered as the method name?

**Comment by odersky** [comment]
I am also not sure what's the best name. I have the feeling that `as` is already used quite a lot for specific conversions. I originally wa thinking of `inject`, that's another possibility. `into` comes from Rust.

One consideration is that the name should work well with and without a type parameter. Compare:
```scala
xs.into            xs.as            xs.inject
xs.into[C]       xs.as[C]      xs.inject[C]
```
I have the impression `inject` works  best without a type parameter, `into` is sort of OK, but `as` is taking it too far.

**Comment by SethTisue** [comment]
how about `convert` or `converted`?

**Comment by odersky** [comment]
`convert` would be the most obvious choice. That's often the best one, I agree.

**Comment by soronpo** [comment]
Maybe `convertTo[T]` will be a little bit more clear, although more verbose.

File: library/src/scala/Conversion.scala [hunk 1]
```diff
@@ -17,7 +17,16 @@ package scala
...
+*  the `into` extension method.
 */
 @java.lang.FunctionalInterface
 abstract class Conversion[-T, +U] extends Function1[T, U]:
+  /** Convert value `x` of type `T` to type `U` */
   def apply(x: T): U
+object Conversion:
+
+  extension [T](x: T)
+    /** `x.into[U]` converts a value `x` of type `T` to type `U` */
```
Comment (rjolly) [inline library/src/scala/Conversion.scala:31]: Need to change comment too

**Review by rjolly — APPROVED** [review]

**Pull request merged**